cdap-data-fabric/src/main/java/co/cask/cdap/data2/sql/PostgresSqlStructuredTable.java

```diff
@@ -204,14 +205,19 @@ public void increment(Collection<Field<?>> keys, String column, long amount)
     }
     fieldValidator.validatePrimaryKeys(keys, false);
 
-    String sql = getIncrementStatement(keys, column);
+    List<Field<?>> fieldsWithValue = new ArrayList<>(keys);
+    // If the row does not exist, insert it with long field = amount
+    fieldsWithValue.add(Fields.longField(column, amount));
+    String sql = getWriteSqlQuery(fieldsWithValue, column);
     try (PreparedStatement statement = connection.prepareStatement(sql)) {
       statement.setLong(1, amount);
-      int index = 2;
-      for (Field<?> key : keys) {
+      int index = 1;
```

Comment:
doesn't this overwrite what you set in line 213?

Reply:
Yes thanks, good catch. We populate the amount after the loop - this was an old implementation :)